lib/client/ClientRenderingWorkflow.js

```diff
@@ -172,6 +168,22 @@ var ClientRenderingWorkflow = {
 
 };
 
+function promisify(method) {
```

Comment:
it turned out new Promise would be a little more complicated because the return values of the methods can be either a value or Promise of a value. also complicated was how to handle errors. if i called method() directly inside the new Promise and it threw an error, the error would not be caught and call the reject callback. using try/catch i could do that, but i figured that a promise chain handles all these complexities for me